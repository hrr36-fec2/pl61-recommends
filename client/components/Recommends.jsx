import axios from 'axios';

class Recommends extends React.Component {
  constructor() {
    super();
    this.state = {
      tracks: []
    }
  }

  componentDidMount() {
    this.getRecommends();
  }

  getRecommends() {
    axios.get('http://127.0.0.1:3003/recommends')
      .then(res => {
        this.setState({ tracks: res.data });
      })
      .catch(err => { console.log(err); });
  }

  render() {
    return (
      <ul>
        {this.state.tracks.map((ele, i) => {
          return (
            <li key={i}>
              <div>{ele.title}</div>
              <div>{ele.artist + '\ \u00B7\ ' + ele.album}</div>
            </li>
          )
        })}
      </ul>
    );
  }
}

export default Recommends;
