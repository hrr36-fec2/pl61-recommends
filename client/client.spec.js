import React from 'react';
import { shallow } from 'enzyme';
import 'babel-polyfill';

import Recommends from './components/recommends';

describe('recommends', () => {
  it('should render', () => {
    const component = shallow(<Recommends />);

    expect(component).toMatchSnapshot();
  });

  it('should something', () => {
    const wrapper = shallow(<Recommends />);
    const instance = wrapper.instance();

    jest.spyOn(instance, 'handleToggle');
    instance.handleToggle();

    expect(instance.handleToggle).toHaveBeenCalled();
  });
});
