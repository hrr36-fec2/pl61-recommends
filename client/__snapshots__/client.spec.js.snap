// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recommends should render 1`] = `
<styles__Container>
  <styles__Header>
    <div>
      <styles__Toggle>
        <styles__Flex
          onClick={[Function]}
        >
          <styles__Title>
            Recommended Songs
          </styles__Title>
          <styles__Arrow
            className="fas fa-caret-down"
          />
        </styles__Flex>
      </styles__Toggle>
      <styles__Info>
        Songs that you might like.
      </styles__Info>
    </div>
    <styles__Refresh
      fade={false}
      onClick={[Function]}
    >
      REFRESH
    </styles__Refresh>
  </styles__Header>
  <Tracks
    added={-1}
    fade={false}
    playing={-1}
    removeTrack={[Function]}
    resetAdded={[Function]}
    tracks={Array []}
    updatePlay={[Function]}
  />
</styles__Container>
`;
